### PR TITLE
Trying to add some filters

### PR DESCRIPTION
**Pull request by muschellij2**
Trying to add some filters.  But 
1) Naming isn't great
2) I'm not sure these are in other functions
3) I am getting symbol not found errors.  I will re-install ITKR/ANTsRCore again from scratch to see if that is still the case.

Let me know if there is something wrong with with the cpp files.  I'm very rusty with this and haven't used ITK headers before so probably pretty bad but tried to adapt `smoothImage` from `ANTsRCore` to do the job.

### CODE REVIEW

**Comment by muschellij2** [comment]
Hey @stnava, I can install the package and run the filter commands I made, but when I run `R CMD check` I get:

```
Error: package or namespace load failed for ‘ANTsR’ in dyn.load(file, DLLpath = DLLpath, ...):
 unable to load shared object '/Users/johnmuschelli/Dropbox/Packages/ANTsR.Rcheck/ANTsR/libs/ANTsR.so':
  dlopen(/Users/johnmuschelli/Dropbox/Packages/ANTsR.Rcheck/ANTsR/libs/ANTsR.so, 6): Symbol not found: _confidenceConnectedImage
  Referenced from: /Users/johnmuschelli/Dropbox/Packages/ANTsR.Rcheck/ANTsR/libs/ANTsR.so
  Expected in: flat namespace
 in /Users/johnmuschelli/Dropbox/Packages/ANTsR.Rcheck/ANTsR/libs/ANTsR.so
```
I have compiled ITKR, ANTsRCore and ANTsR all using the same architecture and compilers.  Do you have any idea why this would be?  I am new to working in your C++ ecosystem with this package.  I have added `confidenceConnectedImage` to `init.c`.  Do I need to do anything else to make sure it's "exported"?

I'm at a loss

**Comment by stnava** [comment]
see @ntustison PR https://github.com/ANTsX/ANTsR/pull/213/files for guidance

we should probably add this info / how-to on the wiki


brian

File: src/init.c [hunk 1]
```diff
@@ -42,7 +42,7 @@ static const R_CallMethodDef CallEntries[] = {
     {"antsMotionCorrStats",             (DL_FUNC) &antsMotionCorrStats,              4},
     {"centerOfMass",                    (DL_FUNC) &centerOfMass,                     1},
     {"confidenceConnectedImage",        (DL_FUNC) &confidenceConnectedImage,         6},
-    {"connectedThresholdImage",         (DL_FUNC) &connectedThresholdImage,          5},
+    // {"connectedThresholdImage",         (DL_FUNC) &connectedThresholdImage,          5},
```
Comment (stnava) [inline src/init.c]: shouldnt this be uncommented?

Comment (muschellij2) [reply src/init.c]: No - I was adding both `confidenceConnectedImage` and `connectedThresholdImage` for the PR, but it was errroring so I was doing one at a time.  

I believe I did exactly what @ntustison https://github.com/ANTsX/ANTsR/pull/213/files had done, but I'm unsure why I'd get an error.  Do you think it has anything to do with `include` headers compared to https://github.com/ANTsX/ANTsR/pull/213/files#diff-b28a056446d4b76f1115cd7ec663a226?

Comment (muschellij2) [reply src/init.c]: Still getting the error at https://travis-ci.org/ANTsX/ANTsR/jobs/394584887#L2193

**Comment by muschellij2** [comment]
That may be, but I don't see any efforts to port them really other than here.

That package hasn't been updated in over 2 years so I'm not hopeful that it's going to be maintained or  stable any time soon.

**Comment by muschellij2** [comment]
OK @stnava, I think I know what finally happened.  The example you gave from @ntustison worked because the function he did included `itkHistogramMatchingImageFilter.h`, which is located in the `ANTsRCore` `include` folder.  

Exposure of all of `ITK`, which should be exposed by `ITKR`, would be possible, but I believe the headers must be in the `include` folder after installation, but that only contains `dummy.h`.  The real library is in `libs/include/ITK-5.0`.  This seems to install all of ITK, but not expose the headers, which would enable the real power of that package.  

When I was referencing `itkConnectedThresholdImageFilter.h` from `src/connectedThresholdImage.cpp`, I had assumed that ITK, though ITKR could be linked to, but I think that's incorrect.  It seems as though the only headers available to link to would be in the `ANTsRCore/include` folder.  Is this correct?  If not, I'm unsure as to why my example fails but @ntustison did not, because they are very similarly set up.  The only main difference i can see is that the header he references is in `ANTsRCore/include`, while mine is not, but the ITKR headers should be able to find `itkConnectedThresholdImageFilter.h`.

**Comment by muschellij2** [comment]
Looking more into this, it seems as though all of ITK is included as `configure` in `ANTsRCore` uses `-I\${ITK}` where `ITK` is `ITKR:::itkIncludes()`, so this is how you solve the problem.  It seems as though `ANTsR` gets this as well in it's `PKG_CPPFLAGS` argument, so I don't know what the issue could be then.

**Comment by muschellij2** [comment]
@stnava this should be ready to merge for now - but I still don't understand the discussion points above I had about `PKG_CPPFLAGS` and such.

**Comment by stnava** [comment]
is this ready to merge?

**Comment by stnava** [comment]
ie *still* read to merge?

**Comment by muschellij2** [comment]
This is ready to be integrated.  This must be used with the current (as of right now) version of `ANTsRCore` due to `getCentroids` and other imports.    

This should resolve a good deal of the issues I had and hopefully address a lot of the spots of "randomness" in the functionality and allow users to make more reproducible pipelines.  There are still some areas that need to be tested.  We should run `covr` to see the package coverage and the blind-spots/untested functions.

**Comment by muschellij2** [comment]
These builds need to wait until https://github.com/ANTsX/ANTsRCore builds a new release then travis will work because it's currently pulling a binary from the releases, which doesn't have the necessary version.

**Pull request merged**